src/AppBundle/Entity/Affiliate.php

```diff
@@ -208,19 +215,19 @@ public function addCategory(Category $category)
     /**
      * Remove category
      *
-     * @param \AppBundle\Entity\Category $category
+     * @param Category $category
      */
-    public function removeCategory(Category $category)
+    public function removeCategory(?Category $category)
     {
         $this->categories->removeElement($category);
     }
 
     /**
      * Get categories
      *
-     * @return \Doctrine\Common\Collections\Collection
+     * @return ArrayCollection
      */
-    public function getCategories()
+    public function getCategories() : ?ArrayCollection
```

Comment:
It can be PersistenCollection, which will give you error. In this case better to write:
```php
public function getCategories() : ?Collection
```